drivers/sdhc/rcar_mmc.c

```diff
@@ -0,0 +1,2223 @@
+/*
+ * Copyright (c) 2023 EPAM Systems
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT renesas_rcar_mmc
+
+#include <zephyr/devicetree.h>
+#include <zephyr/drivers/disk.h>
+#include <zephyr/drivers/sdhc.h>
+#include <zephyr/drivers/clock_control/renesas_cpg_mssr.h>
+#include <zephyr/drivers/pinctrl.h>
+#include <zephyr/logging/log.h>
+#include <zephyr/cache.h>
+#include <zephyr/drivers/regulator.h>
+
+#include "rcar_mmc_registers.h"
+
+#define PINCTRL_STATE_UHS PINCTRL_STATE_PRIV_START
+
+/**
+ * @note we don't need any locks here, because SDHC subsystem cares about it
+ */
+
+LOG_MODULE_REGISTER(rcar_mmc, CONFIG_LOG_DEFAULT_LEVEL);
+
+#define MMC_POLL_FLAGS_TIMEOUT_US 100000
+#define MMC_POLL_FLAGS_ONE_CYCLE_TIMEOUT_US 1
+#define MMC_BUS_CLOCK_FREQ 800000000
+/*
+ * SD/MMC clock for Gen3/Gen4 R-car boards can't be equal to 208 MHz,
+ * but we can run SDR104 on lower frequencies:
+ *    "SDR104: UHS-I 1.8V signaling, Frequency up to 208 MHz"
+ * so according to SD card standard it is possible to use lower frequencies,
+ * and we need to pass check of frequency in sdmmc in order to use sdr104 mode.
```

Comment:
This looks a bit hacky for me. Could sdmmc core code be fixed to allow lower frequencies? I believe, this is the proper approach.

Reply:
I completely agree with you that it's a bit hacky. Would you mind if I open an issue for it, and then we can address it in a separate pull request?

Reply:
Actually yes, this is the great idea. You can link this discussion to the new issue.

Reply:
added an issue: https://github.com/zephyrproject-rtos/zephyr/issues/67943